pkg/engine/imageVerify.go

```diff
@@ -148,67 +184,65 @@ type imageVerifier struct {
 
 // verify applies policy rules to each matching image. The policy rule results and annotation patches are
 // added to tme imageVerifier `resp` and `ivm` fields.
-func (iv *imageVerifier) verify(imageVerify kyvernov1.ImageVerification, images map[string]map[string]apiutils.ImageInfo) {
+func (iv *imageVerifier) verify(imageVerify kyvernov1.ImageVerification, imageInfos []apiutils.ImageInfo) {
 	// for backward compatibility
 	imageVerify = *imageVerify.Convert()
 
-	for _, infoMap := range images {
-		for _, imageInfo := range infoMap {
-			image := imageInfo.String()
+	for _, imageInfo := range imageInfos {
+		image := imageInfo.String()
 
-			if !imageMatches(image, imageVerify.ImageReferences) {
-				iv.logger.V(4).Info("image does not match pattern", "image", image, "patterns", imageVerify.ImageReferences)
-				continue
-			}
-
-			if hasImageVerifiedAnnotationChanged(iv.policyContext, iv.logger) {
-				msg := imageVerifyAnnotationKey + " annotation cannot be changed"
-				iv.logger.Info("image verification error", "reason", msg)
-				ruleResp := ruleResponse(*iv.rule, response.ImageVerify, msg, response.RuleStatusFail, nil)
-				iv.resp.PolicyResponse.Rules = append(iv.resp.PolicyResponse.Rules, *ruleResp)
-				incrementAppliedCount(iv.resp)
-				continue
-			}
+		if !imageMatches(image, imageVerify.ImageReferences) {
```

Comment:
do we need to apply this check again? Don't all the images we pass in now, already match the rule?

Reply:
removed check